### ISSUE TITLE
[BUG] workspaces packages are built in parallel after npm ci even though they depend on each other

### ISSUE THREAD

**Issue by dreamorosi** [npm/cli#4475]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I'm using `npm@8.5.2` and making use of the workspaces feature, my (simplified) project's structure is the following:
```sh
my_project
|_ packages
   |_ commons
   |_ logger
   |_ metrics
   |_ tracing
package.json 
```

In my `package.json` file I have the following config:
```json
"workspaces": [
    "packages/*"
 ]
```

The project uses TypeScript and the packages `logger`, `metrics`, `tracing` all depend on `commons`.

When I run `npm ci` in my CI/CD environment (GitHub Actions) after installing the dependencies npm tries to build (`tsc`) the packages but appear to do so in parallel and so the build of all the dependant packages fails because it's being done before `commons` is ready.

### Expected Behavior

Npm to understand that some packages of the workspace are dependant on others and build accordingly.

Alternatively to allow me to specify that I want to use a remote (i.e. not linked) version of the package even though the version number is the same as the local one.

### Steps To Reproduce

Read above, also you can see the error (with `logLevel == silly`) here https://github.com/dreamorosi/aws-lambda-powertools-typescript/runs/5345787297?check_suite_focus=true#step:6:5380 (<- Line that causes the error)

### Environment

- npm: 8.5.2
- Node.js: lts/fermium (aka 14.x)
- OS Name: Ubuntu
- System Model Name: GitHub Actions
- npm config: N/A

**Comment by lukekarrys** [comment]
What package.json script are you using to run `tsc`? If it's `prepare` then a workaround is to use the `--foreground-scripts` flag (or set it in your project `.npmrc`) and manually order the your `"workspaces": []` array to be the order you want them to be run.

There's more info in #3034 and #4100. This might be a duplicate of one of those, but I will live this open in case it's something else.

**Comment by dreamorosi** [comment]
Hello @lukekarrys, thanks for the quick reply, really appreciate it. I wasn't aware that the issue was in the `prepare` script and so I was searching for `install` or `ci` and didn't find the other two issues.

We are indeed using `prepare` in our project. I had already tried to explicitly order the `"workspaces": []` but without `--foreground-scripts` it still failed.

I can confirm that using the combination of both that you suggested works. 

The two action items you mentioned [in this comment](https://github.com/npm/cli/issues/4100#issuecomment-1053006177) are a must in my opinion. Documentation could clarify this behaviour but also a specific flag to control this behaviour would be useful as the link between `--foreground-scripts` and concurrency/parallel execution is not immediately clear, at least for those not familiar with the inner workings of the project.

If you prefer to close this issue as duplicate of #4100 feel free to, I'll subscribe to that one in that case.

**Comment by lukekarrys** [comment]
Thanks for the follow-up @dreamorosi, and I'm glad that workaround was able to resolve your initial problem. I'm going to close this and we can track it the other linked issue.
